### PR TITLE
Fixes #4712

### LINKED ISSUE #4712: Can we add `flake8`, `mypy` and python `black` to this checklist ?
image
**Most pull requests have pre-commit checks failing because the author haven't used it.**

### PR DESCRIPTION
**Pull request by Joe-Sin7h**
### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [X] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
`Fixes #4712` is a really poor title for a pull request because it forces the reviewer to type by hand the issue that is being fixed.  Instead, the title should briefly describe the issue being fixed and the commit message should contain `Fixes #4712` which
1. Gives the reviewer an easy link to click on to read the issues, and
2. Allows GitHub to autoclose the issue when the pull request is landed.

**Comment by Joe-Sin7h** [comment]
> `Fixes #4712` is a really poor title for a pull request because it forces the reviewer to type by hand the issue that is being fixed. Instead, the title should briefly describe the issue being fixed and the _commit message_ should contain `Fixes #4712` which
> 
> 1. Gives the reviewer an easy link to click on to read the issues, and
> 2. Allows GitHub to autoclose the issue when the pull request is landed.

okay got it, thanks for mentioning it.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -14,7 +14,8 @@
 * [ ] All new Python files are placed inside an existing directory.
 * [ ] All filenames are in all lowercase characters with no spaces or dashes.
 * [ ] All functions and variable names follow Python naming conventions.
-* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
+* [ ] All function parameters and return values are annotated with Python [mypy](https://docs.python.org/3/library/typing.html).
```
Comment (cclauss) [inline .github/pull_request_template.md:17]: Please undo.  These are type hints.  Mypy and Pyright are both programs that test type hints.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -14,7 +14,8 @@
...
 * [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
 * [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
+* [ ] Modified/Added code follows [black](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html) and [flake8](https://flake8.pycqa.org/en/latest/) code style.
```
Comment (cclauss) [inline .github/pull_request_template.md:20]: We have tests for this so we do not need to ask the contributor if they have done this.

**Pull request closed without merging**